src/octree/generation.rs

```diff
@@ -296,18 +289,24 @@ pub fn build_octree_from_file(
 pub fn build_octree(
     output_directory: impl AsRef<Path>,
     resolution: f64,
-    bounding_box: Aabb3<f64>,
+    bounding_box: Aabb<f64>,
     input: impl Iterator<Item = PointsBatch> + NumberOfPoints + Send,
     attributes: &[&str],
 ) {
     attempt_increasing_rlimit_to_max();
 
-    // TODO(ksavinash9): This function should return a Result.
+    let attribute_data_types = vec![
```

Comment:
Made a small orthogonal change here to be more explicit than `Default::default()`. I'm fine with reverting it if you prefer.